docs/int_header/INT_header.md

```diff
@@ -9,83 +9,61 @@ These definitions contain modifications and required fields for Transparent Secu
 
-The INT header is defined in two portions.  One is the Header for the INT metadata and the second is the actual metadata.  For both of these, we follow the INT specification with the addition of two bitmasks.  This additional value, an 8 octet value for the originating device ID will be proposed to be added to the INT specification.
+Please refer to the 2.0 draft or release version of the P4 Application for the metadata header format.  This document includes the definition of the domain specific extension and examples using a TCP/UDP header to encapsualte the packet.
 
-Please refer to the 2.0 draft or release version of the P4 Application for the metadata header format.  This document includes the definition of the additional metadata types for the originating device and a proposed IP shim.
-
-The other shim types, including UDP/TCP, can also be used with the probe marker and not the differentiated service bit (DSCP), but those do not cover the range of IP traffic that is used in DDoS attacks.
-
-## IP Header update
-
-### Encapsulation
-
-We want to preserve the routing and other information on the packet, so we will modify the protocol (IPv4) / next header (IPv6), length and IPv4 the header checksum.
-
-Define a new L3 protocol.  The protocol for the INT header will be 63 "Any local network" since the INT header should never be exposed outside of the service providers network.  There is no specification for protocol 63.
-
-IPv4:
-
-* Update the total length to match the new size
-* Set the protocol to the INT protocol number
-* Update the header checksum
-
-IPv6:
+## INT Header
 
-* Set the Next Header to be the identifier for an IPv6 extension header to the INT protocol number.  If existing extension headers are in place, INT will prepend to the first extension header
-* The INT header next header field will be the existing IPv6 next header value
-* The Hdr Ext Len will be updated appropriately
+### INT metadata header (12 bytes)
 
-### IP header shim (4 bytes)
+The hop-by-hop INT header will follow the header as described in section 4.7. INT Hop-by-Hop Metadata Header Format in the current INT specification.
 
-This shim shall be 4 bytes and include the original protocol (IPv4) or next protocol (IPv6) and the length of the INT metadata.
+* INT instructions are encoded as a bitmap in the 16 bit INT Instruction field.
 
-Type: (1 octet) This field indicates the type of INT Header following the shim header. Two Type values are used: one for the hop-by-hop header type and the other for the destination header type.
+  * bit0: 4 octet Switch ID which is unique across the network
 
-Reserved: (1 octet)
+### Per-Hop INT Metadata record (12 bytes)
 
-Length: (1 octet) This is the total length of INT metadata header, INT stack and the shim header in 4-byte words. A non-INT device may read this field and skip over INT headers.
+This metadata will only contain one record and will not be updated on subsequent hops.
```

Comment:
The example shows each hop adding their switchID?  so it does get updated by subsequent hops (if capable)?

Reply:
Moved this to the domain specific section.